src/query/server/query.go

```diff
@@ -541,6 +542,64 @@ func Run(runOpts RunOptions) RunResult {
 		if err != nil {
 			logger.Fatal("unable to setup downsampler for prom remote backend", zap.Error(err))
 		}
+	case config.DualStorageType:
```

Comment:
Can we do `fallthrough` to avoid code duplication?
```
case config.DualStorageType:
  // Set up prom
   fallthrough
case "", config.M3DBStorageType,config.DualStorageType:
  // Set up m3db
```

Reply:
emm, i don't think we can easily do it in this case, dual storage needs to composite two storages using `backendStorage = composite.Compose(logger, m3Storage, promRemoteStorage)` and then pass it into downsampler, while in m3db, we don't have that.

Reply:
Actually I can use `fallthrough` but a bit different syntax, let me update and test this out.